### ISSUE TITLE
[feature request] implement connection services

### ISSUE THREAD

**Issue by itcarroll** [tomoakin/RPostgreSQL#90]
Quoting from the [current docs](https://www.postgresql.org/docs/current/static/libpq-pgservice.html):

    The connection service file allows libpq connection parameters to be associated with
    a single service name. That service name can then be specified by a libpq connection,
    and the associated settings will be used.

The implementation should allow creating a connection with

```
dbConnect(PostgreSQL(), service = 'xyz')
```

where the (by default) `~/.pg_service.conf` file contains an entry for `xyz` that centrally stores all the connection parameters.

Happy to start work on a PR.

**Comment by itcarroll** [comment]
Oh, loophole!

    dbConnect(PostgreSQL(), dbname = 'postgresql://@/?service=xyz')

Achieves what I wanted, if inelegantly. I still think it would be good to provide the `service` parameter as its own argument, so I'll leave my issue open.

**Comment by AmyMikhail** [comment]
Not sure if this is referring to RStudio preview versions >= 1.2.1?

I have installed the preview version and was excited to try out the new SQL database 'connection' feature, only to discover that this has to be set up for each package that makes connections to databases and isn't yet automatically detected for `RPostgreSQL`.

The FAQ for the preview version says that package authors need to implement the [RStudio connections contract](https://rstudio.github.io/rstudio-extensions/connections-contract.html).

Would it be possible to implement this for the `RPostgreSQL` package?  This new feature in RStudio would be really useful for creating queries.

**Comment by itcarroll** [comment]
No, this feature request isn't about RStudio "connections". I'd suggest you
open a separate issue, or switch to the RPostgres package (which is what I
did).
